tls/s2n_ktls_io.c

```diff
@@ -405,3 +406,35 @@ int s2n_sendfile(struct s2n_connection *conn, int in_fd, off_t offset, size_t co
     *blocked = S2N_NOT_BLOCKED;
     return S2N_SUCCESS;
 }
+
+int s2n_ktls_read_full_record(struct s2n_connection *conn, uint8_t *record_type)
```

Comment:
Yeah that'll be part of the application data s2n_recv PR. The short answer is that it will read application data if there's application data available instead of a control message, so we better have already read any application data :)

But we actually could implement s2n_recv with just this-- see my note in the testing section of the description. It's just inefficient because we copy data from socket->buffer->output instead of socket->output.